oapi_validate.go

```diff
@@ -51,12 +64,22 @@ func OapiRequestValidatorWithOptions(swagger *openapi3.T, options *Options) func
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 
 			// validate request
-			if statusCode, err := validateRequest(r, router, options); err != nil {
-				if options != nil && options.ErrorHandler != nil {
-					options.ErrorHandler(w, err.Error(), statusCode)
-				} else {
-					http.Error(w, err.Error(), statusCode)
+			if statusCode, route, err := validateRequest(r, router, options); err != nil {
+				if options != nil {
+					if options.ErrorHandlerWithOpts != nil {
+						options.ErrorHandlerWithOpts(w, err.Error(), statusCode, ErrorHandlerOpts{
```

Comment:
If change to `ErrorHandlerOpts` is implemented then we could add here:
```suggestion
						options.ErrorHandlerWithOpts(w, err.Error(), statusCode, ErrorHandlerOpts{
						Error: err,
```

Reply:
I made that change to the PR. Please review the PR to make sure I correctly understood what you were suggesting.